src/filters/matches.rs

```diff
@@ -0,0 +1,159 @@
+crate::include_proto!("quilkin.extensions.filters.matches.v1alpha1");
+
+use self::quilkin::extensions::filters::matches::v1alpha1::Matches as ProtoConfig;
+use crate::filters::{prelude::*, registry::FilterRegistry};
+
+pub const NAME: &str = "quilkin.extensions.filters.matches.v1alpha1.Matches";
+
+/// Creates a new factory for generating capture filters.
+pub fn factory() -> DynFilterFactory {
+    Box::from(MatchesFactory::new())
+}
+
+struct Matches {
+    config: Config,
+    metrics_registry: prometheus::Registry,
+    filter_registry: FilterRegistry,
+}
+
+impl Matches {
+    fn new(
+        config: Config,
+        filter_registry: FilterRegistry,
+        metrics_registry: prometheus::Registry,
+    ) -> Self {
+        Self {
+            config,
+            filter_registry,
+            metrics_registry,
+        }
+    }
+}
+
+/// The actual behaviour of the matches filter, the only thing that changes
+/// between read and write is which config and direction is called.
+macro_rules! matches_filter {
+    ($this:ident, $context:ident, $direction:ident, $config:ident) => {{
+        let ctx = $context;
+        let config = match $this.config.$config.as_ref() {
```

Comment:
My reasoning was that it's optional for filters where you just want the default behaviour.

Reply:
there really isn't a reasonable default behavior for this filter? In that if the filter is to do something then it needs some config, here the default behavior seems to be the same as not having the filter at all which tells me it isn't really a behavior of the filter

Reply:
Sorry, I meant a default for another filter. For this filter it's optional because you might want to match only in one direction `on_read` or `on_write`, or both. You're right that there's no reasonable default for having neither of those specified.